CHANGELOG.md

```diff
@@ -10,6 +10,9 @@ internal API changes are not present.
 Main (unreleased)
 -----------------
 
+v1.2.0-rc.0
```

Comment:
Heads up if you do that in this PR we can’t backport this to the release branch as it has other changes the release branch shouldn’t have. Not a big issue since you need to create two PRs either way

Reply:
We can't backport any of the changes to the release branch anyway:
* [release/v1.2](https://github.com/grafana/alloy/tree/release/v1.2) isn't meant to have `Main (unreleased)` in the changelog.
* [release/v1.2](https://github.com/grafana/alloy/tree/release/v1.2) isn't meant to have its `VERSION` and `docs/sources/_index.md` files updated.